test/integration_test.py

```diff
@@ -1037,3 +1053,93 @@ def test_local_filtered_index_endpoints(self):
                                              catalog=self.catalog)).url
         response = requests.get(url)
         self.assertEqual(200, response.status_code)
+
+
+class CanBundleScriptIntegrationTest(IntegrationTestCase):
+
+    def _test_catalog(self, catalog: config.Catalog):
+        fqid = self.bundle_fqid(catalog.name)
+        log.info('Canning bundle %r from catalog %r', fqid, catalog.name)
+        with tempfile.TemporaryDirectory() as d:
+            self._can_bundle(source=str(fqid.source.spec),
+                             uuid=fqid.uuid,
+                             version=fqid.version,
+                             output_dir=d)
+
+            def file_name(part):
+                return f'{fqid.uuid}.{part}.json'
+
+            manifest_file_name = file_name('manifest')
+            metadata_file_name = file_name('metadata')
+            expected_files = sorted([manifest_file_name, metadata_file_name])
+            generated_files = sorted(os.listdir(d))
+            self.assertListEqual(generated_files, expected_files)
+
+            with open(f'{d}/{manifest_file_name}') as f:
+                manifest = json.load(f)
+            with open(f'{d}/{metadata_file_name}') as f:
+                metadata = json.load(f)
+
+            self.assertIsInstance(manifest, list)
+            self.assertIsInstance(metadata, dict)
+
+            manifest_files = sorted(e['name'] for e in manifest if e['indexed'])
+            metadata_files = sorted(metadata.keys())
+
+            if catalog.plugins['repository'] == 'canned':
+                # FIXME: Manifest entry not generated for links.json by
+                #        StagingArea.get_bundle
+                #        https://github.com/DataBiosphere/hca-metadata-api/issues/52
+                assert 'links.json' not in manifest_files
+                metadata_files.remove('links.json')
+
+            self.assertListEqual(manifest_files, metadata_files)
+
+    def test_can_bundle_configured_catalogs(self):
+        for catalog_name, catalog in config.catalogs.items():
+            if catalog.is_integration_test_catalog:
+                with self.subTest(catalog=catalog.name,
+                                  repository=catalog.plugins['repository']):
+                    self._test_catalog(catalog)
+
+    canned_repo = 'https://github.com/HumanCellAtlas/schema-test-data/tree/master/tests'
+
+    @mock.patch.dict(os.environ,
+                     azul_canned_sources=f'{canned_repo}:')
+    @mock.patch.object(azul.Config, 'catalogs', new_callable=PropertyMock)
+    # Must be mocked independently due to caching
+    @mock.patch.object(azul.Config, 'default_catalog', new_callable=PropertyMock)
+    def test_can_bundle_canned_repository(self, mock_default_catalog, mock_catalogs):
```

Comment:
I think this might look more coherent if you used `with` statements for patching. Please also consider resetting any cached properties instead of mocking them separately. See `azul.caching.CachedProperty` for details.